packages/protocol/src/protocol.yml

```diff
@@ -1191,6 +1191,22 @@ BrowserContext:
       returns:
         writableStream: WritableStream
 
+    createTempDirectory:
+      parameters:
+        root: string
+        items:
+          type: array
+          items:
+            type: object
+            properties:
+              name: string
+              lastModifiedMs: number?
+      returns:
+        dir: string
```

Comment:
Why does a client need to know a local path on the server?

Reply:
it's a bit of a hack. We stream 3 files to the server (in the directory). The files itself could live under on the server:

- /tmp-random/my-dir/sub-dir/1.txt
- /tmp-random/my-dir/2.txt
- /tmp-random/my-dir/sub-dir/whatever/3.txt

So the server to this moment does not know what the root directory is. Instead we then send 3 files to the server over our protocol, get `/tmp-random/my-dir` back and pass it as a local-path to the server.

Reply:
Yeah, we should not pass `localPaths` when talking to the server. Let's make server's temp files streams know what they need to.